src/timeline-window.ts

```diff
@@ -135,13 +135,14 @@ export class TimelineWindow {
         // which is important to keep room-switching feeling snappy.
         if (this.timelineSet.getTimelineForEvent(initialEventId)) {
             initFields(this.timelineSet.getTimelineForEvent(initialEventId));
-            return Promise.resolve();
+            return;
         } else if (initialEventId) {
-            return this.client.getEventTimeline(this.timelineSet, initialEventId)
+            await this.client.getEventTimeline(this.timelineSet, initialEventId)
                 .then(initFields);
+            return;
```

Comment:
The function signature is `Promise<void>` here. We shouldn't return anything here.